biosimulations/apps/dispatch-service/src/app/app.controller.ts

```diff
@@ -130,15 +144,39 @@ export class AppController {
                     });
                   });
                 });
+              })
+              .on('error', (err) => {
+                console.log('Error occured in file writing', err);
               });
           }
         }
-      })
-
-      
+      });
     }
   }
 
+  @MessagePattern('dispatch_log')
+  async dispatchLog(data: any) {
+    const slurmjobId = data['hpcOutput']['stdout'].match(/\d+/)[0];
+    const simDirSplit = data['simDir'].split('/');
+    const uuid = simDirSplit[simDirSplit.length - 1];
+    // TODO: research more for better duration
+    this.jobMonitorCronJob(slurmjobId, uuid, 30);
+  }
+
+  async jobMonitorCronJob(jobId: string, uuid: string, seconds: number) {
+    const job = new CronJob(`${seconds.toString()} * * * * *`, async () => {
+      const squeueRes = await this.hpcService.squeueStatus(jobId);
+      const jobMatch = squeueRes.stdout.match(/\d+/);
+      const isJobRunning = jobMatch !== null && jobMatch[0] === jobId;
+      if (!isJobRunning) {
+        this.messageClient.emit('dispatch_finish', { jobId, uuid });
```

Comment:
Not sure if this in elsewhere, but we need to end the job at this point correct?

Reply:
I am a bit confused here. Are you intending to use the message system to trigger a call within the same service?

Reply:
Yes, instead of passing the control of job callback to another method, I thought it'd be better to just notify NATS and end the flow of control of job there only. Afterwards, whichever controller wants to get notified when any simulation finishes, it can just subscribe to this message. 
For now, we only have 1 subscription for this message to start processing the results into JSON.